ConsulAndServiceDiscovery/README.md

```diff
@@ -0,0 +1,114 @@
+# Introduction
+
+This document describes the consul usage and service discovery guidelines. This
+document is sure to evolve as we gain more experience with running consul in
+production. If you have any questions please use the Wikia #consul chat channel.
+
+## Introduction to Consul and Service Discovery
+
+For an introduction to consul please read the [documentation
+provided](https://www.consul.io/intro/) by the creators of consul. Read that
+first.
+
+Services can be "discovered" via DNS or the HTTP API. As an example, consider a
+query for a discussion service node. When using the DNS interface one would
+query for `discussion.service.sjc.consul`. Below is an example:
+
+```
+dig @localhost -p 8600 discussion.service.sjc.consul
```

Comment:
From experience I'd suggest that every service should at least expose one tag and 99.99% of queries should use tag. Because otherwise it is much more difficult to introduce another differently tagged instance of that service.